cmd/krew/cmd/list.go

```diff
@@ -24,11 +24,89 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/kubectl/pkg/util/slice"
 
+	"sigs.k8s.io/krew/pkg/constants"
 	"sigs.k8s.io/krew/pkg/installation"
 )
 
+type Entry struct {
+	Name    string
+	Version string
+}
+
+type Entries []Entry
+
+func sortByName(e Entries) Entries {
+	sort.Slice(e, func(a, b int) bool {
+		return e[a].Name < e[b].Name
+	})
+	return e
+}
+
+// Consume produces a junk GroupVersionKind for obj.GetObjectKind().GroupVersionKind().Empty() check to eat in PrintObj()
+type Consume struct {
+	APIVersion string
+	Kind       string
+}
+
+func (c Consume) SetGroupVersionKind(kind schema.GroupVersionKind) {
+	c.APIVersion, c.Kind = kind.ToAPIVersionAndKind()
+}
+
+func (c Consume) GroupVersionKind() schema.GroupVersionKind {
+	return schema.FromAPIVersionAndKind(c.APIVersion, c.Kind)
+}
+
+func (e Entries) GetObjectKind() schema.ObjectKind {
+	return Consume{constants.CurrentAPIVersion, constants.PluginKind}
```

Comment:
Here, you are re-using the version/kind of the krew manifest. As you are using the k8s machinery, you should not do this here, because the object you are serializing is _not_ the krew manifest. It's ok to reuse the api version, but please use a different kind.

Reply:
The version/kind is being used arbitrarily here and could just as easily be substituted with "_"s, which is what I did in the first place. @ahmetb suggested I use the constants.